app/controllers/ReportController.java

```diff
@@ -101,17 +165,21 @@ public Result query(
      * @return Matching report.
      */
     public Result get(final String id) {
-        return notFound();
-    }
-
-    /**
-     * Delete a specific report.
-     *
-     * @param id The identifier of the report.
-     * @return No content if successful, otherwise an HTTP error code.
-     */
-    public Result delete(final String id) {
-        return notFound();
+        final Organization org;
+        try {
+            org = _organizationRepository.get(request());
+        } catch (final NoSuchElementException e) {
+            return notFound();
```

Comment:
The method above combines the org lookup with the add or update returning a 5xx if the org lookup fails. Here you are returning a 4xx if the org is not found, which can be ambiguous since you are looking for a report. I think you should separate the different error paths above, and here probably map no organization to a 5xx and no report to a 4xx. @BrandonArp ?

Reply:
I would rather change the other path to explicitly return 404 to match this and include a message with not found indicating the missing entity. It seems incorrect to return 5XX when the request is at fault.